src/Azure/Orleans.Streaming.AzureStorage/Providers/Streams/AzureQueue/AzureQueueStreamBuilder.cs

```diff
@@ -29,6 +30,15 @@ public SiloAzureQueueStreamConfigurator(string name, ISiloHostBuilder builder)
         public SiloAzureQueueStreamConfigurator<TDataAdapter> ConfigureAzureQueue(Action<OptionsBuilder<AzureQueueOptions>> configureOptions)
         {
             this.Configure<AzureQueueOptions>(configureOptions);
+            this.Configure<AzureQueueOptions>(b => b.PostConfigure<IOptions<ClusterOptions>>((op, clusterOp) =>
+            {
+                if (op.QueueNamePrefix == null)
+                {
+                    var serviceId = clusterOp.Value?.ServiceId;
+                    var defaultQueueNamePrefix = serviceId + "-" + this.name;
+                    op.QueueNamePrefix = defaultQueueNamePrefix;
```

Comment:
I don't think we should handle this override in the configuration stage.  It makes it difficult to override the behavior.

I'd think what we'd want is for the AzureQueueAdapterFactory to take a concrete list of the queue names that we build in the static Create function using the QueueNamePrefix, serviceId, provider name, and HashRingStreamQueueMapperOptions.TotalQueueCount.  If the patterns we support aren't what a developer needs, they'd need only write their own static Create function and construct the queue name list any way they want (for instance, providing a list of pre-existing queues they want to process events from).

Reply:
I don't think we should handle this override in the configuration stage. It makes it difficult to override the behavior.

What behavior you want to override here? And why exactly `AzureQueueAdapterFactory` is the better place to configure the azure queue name, instead of the AzureQueueOptions. I mean I can argue the same for `AzureTableStorageOptions.TableName`, why is the table name in the AzureTableStorageOptions, instead of the `AzureTableGrainStorageFactory `